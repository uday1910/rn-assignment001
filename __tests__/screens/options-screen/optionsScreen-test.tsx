import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react-native';
import configureStore from '../../../source/store/store';
import OptionsScreen from '../../../source/screens/options-screen/optionsScreen';

describe('AddTodo component test', () => {
  it('InputText render properly according to the snapshot', () => {
    const screen = renderer
      .create(
        <Provider store={configureStore}>
          <OptionsScreen />
        </Provider>,
      )
      .toJSON();
    expect(screen).toMatchSnapshot();
  });

  it('displays the passed-in name', () => {
    const {getByTestId} = render(
      <Provider store={configureStore}>
        <OptionsScreen />
      </Provider>,
    );
    expect(getByTestId('welcome-text')).not.toBeNull();
  });
});
