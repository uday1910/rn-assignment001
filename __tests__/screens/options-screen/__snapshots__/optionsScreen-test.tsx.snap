// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo component test InputText render properly according to the snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "margin": 20,
      }
    }
  >
    <Text
      numberOfLines={0}
      style={
        Object {
          "color": "#404040",
          "fontSize": 22,
          "fontWeight": "bold",
          "textAlign": "right",
        }
      }
      testID="welcome-text"
    >
      Welcome, 
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginVertical": 50,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": "85%",
        }
      }
      testID="submit-button"
    >
      <View
        style={
          Array [
            Object {
              "elevation": 4,
              "shadowColor": "#404040",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.5,
              "shadowRadius": 10,
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#3498DB",
              "borderRadius": 10,
              "height": 50,
              "justifyContent": "center",
              "marginVertical": 10,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          GO TO BUTTON VARIATION
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": "85%",
        }
      }
      testID="submit-button"
    >
      <View
        style={
          Array [
            Object {
              "elevation": 4,
              "shadowColor": "#404040",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.5,
              "shadowRadius": 10,
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "#3498DB",
              "borderRadius": 10,
              "height": 50,
              "justifyContent": "center",
              "marginVertical": 10,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          CHECK RUNNING DEVICE
        </Text>
      </View>
    </View>
  </View>
</View>
`;
